src/routes/routeUtil.ts

```diff
@@ -15,3 +16,39 @@ export const locationFromDefaultOrLogin = (location: Location): boolean => {
   // Page was loaded from a login redirect
   return !!location.state?.fromLoginRedirect;
 };
+
+export const PREV_SEARCH_KEY = 'prev';
+export const prevSearchParam = () =>
+  `${PREV_SEARCH_KEY}=${encodeURIComponent(
+    window.location.pathname + window.location.search
+  )}`;
+
+export const backToFromSearchParam = (
```

Comment:
🚨 this seems to be working great, but just want to flag it is new and kinda complicated. Constructs a `To` based on the `prev` search param if its present, with the ability to inject search params.

Reply:
This is cool! I would just say that the name `backToFromSearchParams` adds some cognitive load, haha! Even after understanding that `To` is the type of the object being returned, I had to concentrate to parse this, I think due to the `to` and `from` words right next to each other in the name. Also I wasn't sure `back` is needed in the name, maybe it's a synonym for `prev` and we should use that instead for consistency? maybe `prevUrlFromSearchParam` is the clearest, but on the other hand, that doesn't indicate that you are returning a type `To` instead of a type `URL`. Maybe `toPrevUrlFromSearchParam` although that's getting verboser.

Either way I also think it would be good to add a comment similar to what you've written here in gh, explaining what this is doing since the name is likely to end up a bit hard to parse either way

Reply:
Cool,  thanks both. I updated to use react-router-dom state for the household navigation piece: https://github.com/greenriver/hmis-frontend/pull/676/commits/ec0e90833ef288efaaec4c0ba295cf5e1af64f9e